src/api/models/lines.js

```diff
@@ -52,16 +52,79 @@ lineSchema.methods.logEvent = function() {
 // update status of line
 // used for manually updating status, instead of using api route
 // also creates log event.
+// returns 'success' upon success, and an error message otherwise.
 lineSchema.methods.updateStatus = function(status) {
-    console.log("Line: Manually Updating Status: " + status);
+    var msg = null;
+    //verify that line progression is happening in order
+    switch (status) {
+
+        case 'notification':
+            if (this.status !== 'preline') {
+                msg = "LineUpdateError: Cannot move to notification status from status " + status;
+            }
+            break;
+        case 'inline':
+            if (this.status !== 'notification') {
+                msg = "LineUpdateError: Cannot move to inline status from status " + status;
+            }
+            break;
+        case 'startrecruiter':
+            if (this.status !== 'inline') {
+                msg = "LineUpdateError: Cannot move to startrecruiter status from status " + status;
+            }
+            break;
+        case 'finishrecruiter':
+            if (this.status !== 'startrecruiter') {
+                msg = "LineUpdateError: Cannot move to finishrecruiter status from status " + status;
+            }
+            break;
+        case 'timeoutchurn':
+            if (this.status !== 'notification' && this.status !== 'inline' && this.status !== 'startrecruiter') {
+                msg = "LineUpdateError: Cannot time out when in status " + status;
+            }
+            break;
+        case 'preline':
+            msg = "Cannot change status to preline - exit and reenter this line";
+            break;
+        case 'voluntarychurn':
+            break; //can enter or exit at any time
+        default:
+            msg = "Invalid status desired: " + status;
+            break;
+    }
+
+    if (msg !== null) {
+        return msg;
+    }
+
     this.status = status;
     this.updated_by = new Date();
     this.save(function(err, line) {
         if (err)
```

Comment:
i think you need to have {} around this if block, otherwise you're returning early.

Reply:
this is intended. msg is set to null at the beginning. Then the verification status checks happen. If any of them fail, msg gets set. So we want to return right away, since an invalid status update was requested